src/Android/Xamarin.Android/Samples/Data/StatsQueryGroupAndSort/StatsQueryGroupAndSort.cs

```diff
@@ -0,0 +1,1000 @@
+
+            // Get results formatted as a dictionary (group names and their associated dictionary of results)
+            var resultsLookup = statQueryResult.ToDictionary(r => string.Join(", ", r.Group.Values), r => r.Statistics);
+
+            // Create an instance of a custom list adapter that has logic to show results as expandable groups
+            var expandableListAdapter = new ExpandableResultsListAdapter(this, resultsLookup);
+
+            // Create an expandable list view and assign the expandable adapter
+            var expandableResultsListView = new ExpandableListView(this);
+            expandableResultsListView.SetAdapter(expandableListAdapter);
+
+            // Show the expandable list view in a dialog
+            AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(this);
+            dialogBuilder.SetView(expandableResultsListView);
+            dialogBuilder.Show();
+        }
+    }
+
+    // Simple class to describe an "order by" option
+    public class OrderFieldOption
+    {
+        // Whether or not to use this field to order results
+        public bool OrderWith { get; set; }
+
+        // The order by info: field name and sort order
+        public OrderBy OrderInfo { get; set; }
+
+        public OrderFieldOption(bool orderWith, OrderBy orderInfo)
+        {
+            OrderWith = orderWith;
+            OrderInfo = orderInfo;
+        }
+    }
+
+    // A class that defines a custom dialog for defining statistics to use in the query
+    public class StatDefinitionsDialog : DialogFragment
+    {
+        // List of field names from the table
+        private List<string> _fieldNames;
+
+        // List of statistic definitions for the query
+        private List<StatisticDefinition> _statisticDefinitions;
+        
+        // Spinner (drop down) to display fields from the table 
+        private Spinner _fieldSpinner;
+
+        // Spinner to display available statistic types (average, sum, maximum, etc.)
+        private Spinner _statSpinner;
+
+        // ListView to show chosen statistic definitions (field name and statistic type)
+        private ListView _statDefListView;
+        
+        public StatDefinitionsDialog(List<string> fieldNames, List<StatisticDefinition> statisticDefs)
+        {
+            // Store field names for the table being queried
+            _fieldNames = fieldNames;
+
+            // Store a list of the current statistic definitions
+            _statisticDefinitions = statisticDefs;
+        }
+
+        public override View OnCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
+        {
+            // Dialog UI to display
+            LinearLayout dialogView = null;
+
+            // Get the context for creating the dialog controls
+            Android.Content.Context ctx = this.Activity.ApplicationContext;
+
+            // Set a dialog title
+            this.Dialog.SetTitle("Statistics Definitions");
+
+            try
```

Comment:
Is this UI code likely to fail? If not, it might be clearer to get rid of the try/catch.